backend/src/test/java/codezap/tag/repository/TemplateTagJpaRepositoryTest.java

```diff
@@ -32,52 +33,114 @@ class TemplateTagJpaRepositoryTest {
+
+        //then
+        assertThat(templateTags).containsExactly(templateTag1, templateTag2);
+    }
+
+    @Test
+    @DisplayName("findDistinctByTemplateIn 조회 테스트")
     void testFindDistinctByTemplateIn() {
         // given
-        List<Long> templateIds = List.of(1L, 2L, 3L);
+        Template template1 = templateRepository.save(
+                new Template(member, "testTemplate1", "testTemplate1", category));
+        Template template2 = templateRepository.save(
+                new Template(member, "testTemplate2", "testTemplate2", category));
+        Template template3 = templateRepository.save(
+                new Template(member, "testTemplate3", "testTemplate3", category));
+
+        Tag tag1 = tagRepository.save(new Tag("tag1"));
+        Tag tag2 = tagRepository.save(new Tag("tag2"));
+        Tag tag3 = tagRepository.save(new Tag("tag3"));
+
+        templateTagRepository.save(new TemplateTag(template1, tag1));
+        templateTagRepository.save(new TemplateTag(template3, tag1));
+        templateTagRepository.save(new TemplateTag(template3, tag3));
 
         // when
-        List<Long> result = templateTagRepository.findDistinctByTemplateIn(templateIds);
+        List<Long> result = templateTagRepository.findDistinctByTemplateIn(
+                List.of(template1.getId(), template3.getId())
+        );
 
         // then
-        assertThat(result).isNotEmpty();
-        assertThat(result.stream().distinct().count())
-                .isEqualTo(result.size());
+        assertThat(result).containsExactly(tag1.getId(), tag3.getId())
+                .hasSize(2);
+    }
+
+    @Test
+    @DisplayName("findAllTemplateIdInTagIds 조회 테스트")
+    void findAllTemplateIdInTagIds() {
+        //given
+        Template template1 = templateRepository.save(new Template(member, "title1", "description1", category));
+        Template template2 = templateRepository.save(new Template(member, "title2", "description2", category));
+        Template template3 = templateRepository.save(new Template(member, "title3", "description3", category));
+
+        Tag tag1 = tagRepository.save(new Tag("tag1"));
+        Tag tag2 = tagRepository.save(new Tag("tag2"));
+        Tag tag3 = tagRepository.save(new Tag("tag3"));
+
+        templateTagRepository.save(new TemplateTag(template1, tag1));
+        templateTagRepository.save(new TemplateTag(template1, tag2));
+        templateTagRepository.save(new TemplateTag(template1, tag3));
+
+        templateTagRepository.save(new TemplateTag(template2, tag2));
+        templateTagRepository.save(new TemplateTag(template2, tag3));
+
+        templateTagRepository.save(new TemplateTag(template3, tag1));
+        templateTagRepository.save(new TemplateTag(template2, tag3));
+
+        //when
+        List<Long> allTemplateIdInTagIds = templateTagRepository.findAllTemplateIdInTagIds(
+                List.of(tag2.getId(), tag3.getId()), 2);
```

Comment:
template 기준으로 개행이 되어있는데, tag 기준으로 개행이 되어있는게 이해하기 더 쉬울 것 같아요~

테스트 목적이 태그에 해당하는 템플릿 아이디 찾기 이기 때문에.. 
```suggestion
        templateTagRepository.save(new TemplateTag(template1, tag1));
        templateTagRepository.save(new TemplateTag(template3, tag1));

        templateTagRepository.save(new TemplateTag(template1, tag2));
        templateTagRepository.save(new TemplateTag(template2, tag2));

        templateTagRepository.save(new TemplateTag(template1, tag3));
        templateTagRepository.save(new TemplateTag(template2, tag3));
                
        //when
        List<Long> allTemplateIdInTagIds = templateTagRepository.findAllTemplateIdInTagIds(
                List.of(tag2.getId(), tag3.getId()), 2);
```
tag 기준으로 개행을 하면 다른 개발자가 확인할 때에도 tag2, tag3 문단의 템플릿 아이디만 확인하면 더 수월하지 않을까요?